### DISCUSSION TITLE
Center a selected node into the viewport

### DISCUSSION THREAD

**Discussion by MatteoGioioso** [xyflow/xyflow#3234] [Q&A] [answered]
I am trying to use `setViewport` to "focus" and center a node in the viewport. However I cannot find the exact position.

Here is a reproducible, as you can see when I try to focus on the node 3 using `positionAbsolute` it does not work.
https://codesandbox.io/s/sweet-hodgkin-djqhw3?file=/App.js

What am I doing wrong?

Thanks

**Comment by chrtze** [comment, marked as answer]
Hey, you can use `fitView` instead and pass the node(s) that you want to center: https://codesandbox.io/s/priceless-gauss-t9k4k5?file=/App.js:1060-1104

**Reply by MatteoGioioso** [reply]
Thanks for that!

**Comment by vinhphanybn1** [comment]
Currently, I want to setViewport for a Node that I selected it. But I'm not sure why it's wrong. Pls help me
if (node?.position) {
      
      const realPosition = screenToFlowPosition(node.position)
      const nodeCenterX = realPosition.x + node.width / 2;
      const nodeCenterY = realPosition.y + node.height / 2;

      setViewport(
        {
          x: nodeCenterX,
          y: nodeCenterY,
          zoom: 1.3,
        },
        { duration: 800 }
      );
    }

**Reply by bcakmakoglu** [reply]
Why not just use `fitView({ nodes: [{ id: nodeId }], duration: 800, minZoom: 1.3, maxZoom: 1.3 })`?
